src/components/message-box/MessageBox.tsx

```diff
@@ -39,53 +59,78 @@ export const MessageBox = ({
   const colorStyle = {
     backgroundColor: theme.static.status[type].background,
   };
+
+  const onPress =
+    onPressConfig &&
+    ('action' in onPressConfig
+      ? onPressConfig.action
+      : () => Linking.openURL(onPressConfig.url));
+
+  const a11yLabel = [title, message, onPressConfig?.text]
+    .filter((s): s is string => !!s)
+    .join(screenReaderPause);
+
   return (
-    <View style={[styles.container, colorStyle, style]}>
+    <TouchableOpacityOrView
+      onClick={onPress}
+      style={[styles.container, colorStyle, style]}
+      accessible={false}
+    >
       {!noStatusIcon && (
         <ThemeIcon
           fill={textColor}
           style={styles.icon}
           svg={messageTypeToIcon(type)}
         />
       )}
-      <View style={styles.content} accessible={true}>
+      <View
+        style={styles.content}
+        accessible={true}
+        accessibilityRole={
+          onPressConfig && ('action' in onPressConfig ? 'button' : 'link')
+        }
+        accessibilityLabel={a11yLabel}
+        accessibilityHint={
+          onPressConfig &&
+          t(MessageBoxTexts.a11yHintPrefix) + onPressConfig.text
+        }
+      >
         {title && (
           <ThemeText
             type="body__primary--bold"
             color={type}
             style={styles.title}
-            accessibilityLabel={title + screenReaderPause}
           >
             {title}
           </ThemeText>
         )}
         <ThemeText color={type} isMarkdown={isMarkdown}>
           {message}
         </ThemeText>
-        {onPress && (
+        {onPressConfig?.text && (
           <ThemeText
             color={type}
-            style={styles.onPressLink}
+            style={styles.linkText}
             type="body__primary--underline"
-            onPress={onPress}
-            accessibilityRole="link"
           >
-            {onPressText ?? t(MessageBoxTexts.tryAgainButton)}
+            {onPressConfig.text}
           </ThemeText>
         )}
       </View>
       {onDismiss && (
-        <TouchableOpacity
-          onPress={onDismiss}
-          accessible={true}
-          accessibilityLabel={t(MessageBoxTexts.dismiss.allyLabel)}
-          accessibilityHint={t(MessageBoxTexts.dismiss.allyHint)}
-          accessibilityRole="button"
-        >
-          <ThemeIcon fill={textColor} svg={Close} />
-        </TouchableOpacity>
+        <View>
+          <TouchableOpacity
+            onPress={onDismiss}
+            accessible={true}
+            accessibilityLabel={t(MessageBoxTexts.dismiss.allyLabel)}
+            accessibilityRole="button"
+            hitSlop={insets.all(theme.spacings.medium)}
+          >
+            <ThemeIcon fill={textColor} svg={Close} />
+          </TouchableOpacity>
+        </View>
```

Comment:
What's the reason for the added `View` tag here?

Reply:
When the `TouchableOpacity` is in a `View` which has `flex: 1` it grows to full height.

With the added `View` / Without the added `View`:
<div>
<img width=350 src="https://user-images.githubusercontent.com/675421/207371639-796d8f19-969c-4b0b-ab81-801792f54ac9.png" />
<img width=350 src="https://user-images.githubusercontent.com/675421/207371633-e245aaa7-a538-49aa-a1f1-8e9b3c7b0d1c.png" />
</div>